sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -7268,23 +7268,29 @@ object functions {
    * @group partition_transforms
    * @since 3.0.0
    */
-  def years(e: Column): Column = withExpr { Years(e.expr) }
+  @deprecated(
+    "functions.years is deprecated. Use functions.partition.years", "3.5.1")
+  def years(e: Column): Column = partition.years(e)
```

Comment:
This seems to be a typo. Shall we use `partitioning` instead of `partition`?
```
[error] /home/runner/work/spark/spark/sql/core/src/main/scala/org/apache/spark/sql/functions.scala:7273:34: not found: value partition
[error]   def years(e: Column): Column = partition.years(e)
```